testsuite/helpers/TsCmp.cc

```diff
@@ -51,19 +56,47 @@ namespace storage
     }
 
 
-    TsCmpActiongraph::TsCmpActiongraph(const Actiongraph::Impl& actiongraph, const Expected& expected)
+    TsCmpActiongraph::TsCmpActiongraph(const string& name)
+    {
+	Environment environment(true, ProbeMode::READ_DEVICEGRAPH, TargetMode::DIRECT);
+	environment.set_devicegraph_filename(name + "-probed.xml");
+
+	Storage storage(environment);
+	storage.get_staging()->load(name + "-staging.xml");
+
+	Actiongraph actiongraph(storage, storage.get_probed(), storage.get_staging());
+	if (access("/usr/bin/dot", X_OK) == 0)
+	{
+	    actiongraph.write_graphviz(name + ".gv", true);
+	    system(("dot -Tpng < " + name + ".gv > " + name + ".png").c_str());
+	}
+
+	TsCmpActiongraph::Expected expected(name + "-expected.txt");
+
+	doit(actiongraph, expected);
+    }
+
+
+    TsCmpActiongraph::TsCmpActiongraph(const Actiongraph& actiongraph, const Expected& expected)
+    {
+	doit(actiongraph, expected);
+    }
+
+
+    void
+    TsCmpActiongraph::doit(const Actiongraph& actiongraph, const Expected& expected)
```

Comment:
doit() is not a good name for anything... what does it do? Let's try to find a more speaking name.